main.go

```diff
@@ -236,6 +245,10 @@ func main() { // nolint:gocyclo
 			mgrOptions.Cache.ByObject[&routev1.Route{}] = cacheLabelConfig
 		}
 
+		if hasGatewayAPI {
```

Comment:
In case of a delayed CRD installation, caching will not get fine-tuned for `HTTPRoutes`. I think we should add a log message saying something along the lines: "Caching fine-tuning is skipped for `HTTPRoutes`, because `gateway.networking.k8s.io` CRD is not currently present in the cluster".